### PR TITLE
Exit better than throw

### PR DESCRIPTION
**Pull request by lucafavatella**
Depends on PR #70.

Before merging this PR, go over this checklist:
- [ ] The issue was approved by a considerable number of votes
- [ ] The rule is listed in the table of contents
- [ ] The rule text respects the structure:
    - [ ] Title
    - [ ] Short Description
    - [ ] Examples
    - [ ] Reasoning
- [ ] It provides links to examples in code for
    - [ ] good scenarios
    - [ ] bad scenarios

### CODE REVIEW

File: src/loud_errors.erl [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
+    something:that(may, fail, WithThis)
   catch
     _:Error ->
       lager:error("Error here: ~p~n"
                   " Arguments: ~p~n"
                   " Stack: ~p", [Error, WithThis, erlang:get_stacktrace()]),
       {error, Error}
   end.
+
+good2(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:good2/1` calls undefined function `something:that/3`

File: src/loud_errors.erl [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
+bad2(WithThis) ->
   try
-    something:that(may, WithThis)
+    something:that(may, fail, WithThis)
+  catch
+    _:Error ->
+      throw({error, Error})
+  end.
+
+good1(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:good1/1` calls undefined function `lager:error/2`

File: src/loud_errors.erl [hunk 3]
```diff
@@ -1,22 +1,38 @@
 -module(loud_errors).
 
--export([bad/1, good/1]).
+-export([bad1/1, bad2/1, good1/1, good2/1]).
 
-bad(WithThis) ->
+bad1(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:bad1/1` calls undefined function `something:that/3`

File: src/loud_errors.erl [hunk 4]
```diff
@@ -1,22 +1,38 @@
...
   try
     something:that(may, fail, WithThis)
   catch
     _:Error ->
       {error, Error}
   end.
 
-good(WithThis) ->
+bad2(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:bad2/1` calls undefined function `something:that/3`

File: src/loud_errors.erl (hunk 2 above)
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:good1/1` calls undefined function `something:that/3`

**Comment by lucafavatella** [comment]
I propose this as I plan to replace several `throw` in sumo.erl (see https://github.com/inaka/erlang_guidelines/pull/70#issuecomment-281972341 for context) with `exit` - that at least has stacktrace. I understand errors shall be more structured (e.g. logging error and returning tuple value tagged with error) but exit appears to me to be good-enough - for sure better than throw.

Feel feel to close straight away if you feed exit not to be good enough.

**Pull request merged**